### PR TITLE
Remove ammonite imports support

### LINKED ISSUE #117: Ammonite-style imports can trigger unused import warnings even when used
```bash
$ touch Main.scala

$ scala-cli compile Main.scala --scala 2.13.6 --scalac-option -Ywarn-unused:imports
Compiling project (Scala 2.13.6, JVM)
Compiled project (Scala 2.13.6, JVM)

$ echo 'import $ivy.`org.typelevel::cats-kernel:2.6.1`;  object Main { println(cats.kernel.Semigroup) }' > Main.scala

$ scala-cli compile Main.scala --scala 2.13.6 --scalac-option -Ywarn-unused:imports 
Compiling project (Scala 2.13.6, JVM)
[warn] ./Main.scala:1:1: Unused import
[warn] ^^^^^^^^^^^^^^
Compiled project (Scala 2.13.6, JVM)

$ scala-cli compile Main.scala --scala 2.13.6 --scalac-option -Ywarn-unused:imports --scalac-option -Xfatal-warnings
Compiling project (Scala 2.13.6, JVM)
[error] ./Main.scala:1:1: Unused import
[error] ^^^^^^^^^^^^^^
Error compiling project (Scala 2.13.6, JVM)
```

launcher version: https://github.com/VirtuslabRnD/scala-cli/releases/download/nightly/scala-cli-x86_64-apple-darwin.gz (downloaded at the time nightly pointed to https://github.com/VirtuslabRnD/scala-cli/commit/c4852a769f7de6534a6e2f4e2110ab2a7b32308e)

system: macOS catalina

### LINKED ISSUE #120: Ammonite imports don't work in REPL
```
$ cat abc.sc 
import $ivy.`org.scalaz::scalaz-core:7.3.5`, scalaz._, Scalaz._
val x = (Option(1) |@| Option(2))(_ + _)
println(x)
$ scala-cli -S 2.12 abc.sc 
Some(3)
```
```
$ scala-cli repl -S 2.12
Welcome to Scala 2.12.14 (OpenJDK 64-Bit Server VM, Java 11.0.11).
Type in expressions for evaluation. Or try :help.

scala> import $ivy.`org.scalaz::scalaz-core:7.3.5`
<console>:11: error: not found: value $ivy
       import $ivy.`org.scalaz::scalaz-core:7.3.5`
```
```
$ scala-cli about
Scala CLI version 0.0.1 (v0.0.1)
```

### LINKED ISSUE #275: Incorrect diagnostics about unused imports when importing `$deps`
### Issue

Today when porting a script I had over to use `scala-cli` I realized that it was giving warnings about unused imports when utilizing a `$dep` import. This happens both in the editor and when using `scala-cli run .` You can reproduce this with the following code snippet:

```scala
// using scala 2.13
// using option -Ywarn-unused

import $dep.`org.mongodb.scala::mongo-scala-driver:4.3.3`
import $dep.`com.outr::scribe-slf4j:3.6.1`

import org.mongodb.scala.MongoClient

object Main extends App {

  val client = MongoClient("someFakeUri")
}
```

In your editor you'll see:

Screenshot 2021-10-24 at 15 52 09

And the same when you run it:

```
❯ scala-cli run .
Compiling project (Scala 2.13.6, JVM)
[warn] ./WakaFetch.scala:4:1: Unused import
[warn] import $dep.`org.mongodb.scala::mongo-scala-driver:4.3.3`
[warn] ^^^^^^^^^^^^^
```

### Expectation

I'd expect this not to warn since I'm using it.

Btw, I'm having tons of fun with this. This is great.

### LINKED ISSUE #1678: Ammonite imports are available in `scala` SIP command.
**Version(s)**
0.1.18

**Describe the bug**
I can use ammonite imports when scala-cli is run under the SIP `scala` command mode

**To Reproduce**
```scala
import $ivy.`org.jsoup:jsoup:1.13.1`

import org.jsoup.Jsoup

@main def foo(url: String) = 
  Jsoup.connect(url).get()
```
compile above with `scala` command

**Expected behaviour**
no special handling for `import $ivy` and `import $dep`

### PR DESCRIPTION
**Pull request by MaciejG604**
Resolves #1678

### CODE REVIEW

File: modules/build/src/main/scala/scala/build/preprocessing/ScalaPreprocessor.scala [hunk 1]
```diff
@@ -234,23 +235,21 @@ case object ScalaPreprocessor extends Preprocessor {
         allowRestrictedFeatures
       ))
 
-    val afterProcessImports: Option[SpecialImportsProcessingOutput] = value {
+    value {
       processSpecialImports(
```
Comment (lwronski) [inline modules/build/src/main/scala/scala/build/preprocessing/ScalaPreprocessor.scala]: The name of this function `processSpecialImports` is misleading as it only generates an error when it encounters ivy or deb dependencies.

File: modules/options/src/main/scala/scala/build/Artifacts.scala [hunk 2]
```diff
@@ -395,7 +371,7 @@ object Artifacts {
         (d, p, a, os.Path(f, Os.pwd))
       },
       extraClassPath,
-      extraCompileOnlyJars ++ extraStubsJars,
+      extraCompileOnlyJars,
```
Comment (lwronski) [inline modules/options/src/main/scala/scala/build/Artifacts.scala:374]: Is `stub` module is not used, we can remove this module from build.sc - [stub-module](https://github.com/VirtusLab/scala-cli/blob/112c02aa0f2029bc0fb7e010cd54da4536e076f1/build.sc#L62)

File: modules/build/src/test/scala/scala/build/tests/BuildTests.scala [hunk 3]
```diff
@@ -260,13 +260,7 @@ abstract class BuildTests(server: Boolean) extends munit.FunSuite {
...
     )
     testInputs.withBuild(defaultOptions, buildThreads, bloopConfigOpt) { (_, _, maybeBuild) =>
-      maybeBuild.orThrow.assertGeneratedEquals(
-        "simple.class",
-        "simple_sc.class",
-        "simple$.class",
-        "simple_sc$.class"
-      )
-      maybeBuild.orThrow.assertNoDiagnostics
+      expect(maybeBuild.isLeft)
```
Comment (lwronski) [inline modules/build/src/test/scala/scala/build/tests/BuildTests.scala:263]: Add some content, that from now `ivy` and `dep` are not supported.  It should be added for every test, when you change `assert`, that now you expected fail with `ivy` and `dep` dependency.

Comment (MaciejG604) [reply modules/build/src/test/scala/scala/build/tests/BuildTests.scala:263]: You mean test names? If so, I changed them.

Comment (lwronski) [reply modules/build/src/test/scala/scala/build/tests/BuildTests.scala:263]: yeah, it better now.

**Review by lwronski — APPROVED** [review]
LGTM

**Review by Gedochao — APPROVED** [review]

**Pull request merged**